### DISCUSSION TITLE
How to use environment variables for field name in DSL?

### DISCUSSION THREAD

**Discussion by YokojimaSkewers** [johnkerl/miller#1204] [General]
What I want to do is equal to

```
mlr put '$'"${FIELD_NAME}"' = hoge'
```

I want to do this with only one string like

```
mlr put '$ENV["FIELD_NAME"] = hoge'
```

but this does not work. Is there a way to do this?

**Comment by aborruso** [comment]
@YokojimaSkewers I think you can use the [system functions](https://miller.readthedocs.io/en/latest/reference-dsl-builtin-functions/index.html#system-functions)

**Reply by YokojimaSkewers** [reply]
Thank you. I didn't know that.

**Comment by johnkerl** [comment]
@YokojimaSkewers do you mean this?

```
$ export FIELD_NAME=bar
$ echo x=abc | mlr put '$[ENV["FIELD_NAME"]] = "hoge"'
x=abc,bar=hoge
```

**Reply by YokojimaSkewers** [reply]
Thank you. This works fine and looks best.

**Reply by aborruso** [reply]
The related documentation https://miller.readthedocs.io/en/latest/reference-dsl-variables/#built-in-variables
